mysqld_exporter.go

```diff
@@ -170,6 +173,76 @@ func init() {
 	prometheus.MustRegister(version.NewCollector("mysqld_exporter"))
 }
 
+func filter(filters map[string]bool, name string) bool {
+	flg := flag.Lookup("collect." + name)
+
+	if flg != nil {
+		if len(filters) > 0 {
+			return flg.Value.(flag.Getter).Get().(bool) && filters[name]
+		}
+		return flg.Value.(flag.Getter).Get().(bool)
+	}
+
+	log.Warnln("Missing collector with name:", name)
+	return false
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	var filters map[string]bool
+	query := r.URL.Query().Get("collect")
+	log.Debugln("collect query:", query)
+
+	if len(query) > 0 {
+		filters = make(map[string]bool)
+		params := strings.Split(query, ",")
```

Comment:
The usage should be `?collect[]=foo&collect[]=bar` to keep in line with other places in Prometheus

Reply:
Good point 👌🏻